run/opencl/ed25519-donna/ed25519-hash.h

```diff
@@ -8,3 +9,44 @@ ed25519_hash(uint8_t *hash, const uint8_t *in, size_t inlen) {
 	SHA512_Update(&ctx, in, inlen);
 	SHA512_Final(hash, &ctx);
 }
+#else
+#include "../opencl_sha2.h"
+
+static void
+ed25519_hash(uint8_t *hash, const uint8_t *in, size_t inlen) {
+	ulong W[16];
+	ulong A, B, C, D, E, F, G, H, t;
+
+	A = SHA2_INIT_A;
+	B = SHA2_INIT_B;
+	C = SHA2_INIT_C;
+	D = SHA2_INIT_D;
+	E = SHA2_INIT_E;
+	F = SHA2_INIT_F;
+	G = SHA2_INIT_G;
+	H = SHA2_INIT_H;
+
+	/* Assume inlen is 32 */
```

Comment:
Is it normally 32 in other (as in future) uses of this? If not, perhaps we should have them as two separate functions, with this being called something like `ed25519_hash32()` and not featuring an inlen argument. We could do so once it hits us but with some bad luck it might take some amount of head scratching before realizing this optimization is the culprit and `inlen` isn't honored at all.

Reply:
Yes, for this curve it will always be 32, so we won't have anything else in this directory. Makes sense to drop the `inlen` argument. I think no need for a separate function nor for a rename, we can just have it without `inlen` and that's it - enough to avoid surprise.